adr/2025_06_13-packages.md

```diff
@@ -0,0 +1,30 @@
+# BlockNote Packages & Sub-Packages
+
+## Core
+
+The `@blocknote/core` package should contain everything needed to build a pure JS editor instance. It has no opinion on the UI layer, and is only concerned with the core editor functionality.
+
+It contains sub-packages for:
+
+- `@blocknote/core/blocks` - Contains the default blocks, inline content, styles and their implementations (excluding things like blockContainer, blockGroup, etc which are core to the editor but should not be part of the schema).
+  - `@blocknote/core/blocks/header` - Contains the header block.
+  - `@blocknote/core/blocks/code-block` - Contains the code block block.
+  - etc...
+- `@blocknote/core/extensions` - Re-exports the extension packages below.
+  - `@blocknote/core/extensions/comments` - Contains the comments extension.
+- `@blocknote/core/exporter` - Contains the exporter package.
+  - `@blocknote/core/exporter/html` - Contains the html exporter package.
+  - `@blocknote/core/exporter/markdown` - Contains the markdown exporter package.
+- `@blocknote/core/server-util` - Contains the server util package.
```

Comment:
not sure if `server-util` should be part of core as it might pull in some other deps?

Reply:
Worthwhile to look into, I should mention that the main reason to split to another package is around deps & install size (or being a very different concern of course)